lib/SPIRVProducerPass.cpp

```diff
@@ -90,6 +90,28 @@ const glsl::ExtInst kGlslExtInstBad = static_cast<glsl::ExtInst>(0);
 
 const char *kCompositeConstructFunctionPrefix = "clspv.composite_construct.";
 
+// SPIRV Module Sections (per 2.4 of the SPIRV spec)
+// These are used to collect SPIRVInstructions by type on-the-fly.
+enum SPIRVSection {
+  kCapabilities,
+  kExtensions,
+  kImports,
+  kMemoryModel,
```

Comment:
Seems a bit odd to define a list for a single instruction. That being said, I'm not sure there's a better alternative.

Reply:
Yes, it's an odd one-off.